markets/storageadapter/ondealsectorcommitted.go

```diff
@@ -96,13 +112,28 @@ func OnDealSectorPreCommitted(ctx context.Context, api getCurrentDealInfoAPI, ev
 		return xerrors.Errorf("failed to set up called handler: %w", err)
 	}
 
+	// If the deal hasn't been activated by the proposed start epoch, timeout
+	// the deal
+	timeoutOnProposedStartEpoch(dealID, proposedDealStartEpoch, eventsApi, func(err error) {
```

Comment:
I am curious if this needs to be a seperate call, or, if you can simply use the timeout functionality of the eventsApi.Called API. Note the fifth function parameter says events.NoTimeout, but this can be a timeout. It may be a duration which would make it potentially not super workable, but getting the current epoch is fairly trival. As I understand it, if the eventsAPI times out, the `called` func will be called but the msg param will be nil -- you can see we actually have a check for that, but it probably is never met cause of the events.NoTimeout parameter. Anyway, this may be worth investigating, since it could simplify the code quite a bit.

Reply:
Ah yeah you're right, that's a better idea 👍